fvm/Equation/DiffusionTerm.set.cpp

```diff
@@ -44,12 +44,15 @@
                 /////////////////////////
                 // RADIUS
                 /////////////////////////
-
                 // Trapping BC: even if the cell is not ignorable, it may still 
                 // be such that the radial flux should be mirrored 
-                if(!isNegativeTrappedRadial && ( Drr(ir, i, j, drr) || Drr(ir+1, i, j, drr) ) ){
+                
+                // For the Ions in DREAM the transient term is defined such that the 
+                // diffusion coefficients should must be negative to get the correct
+                // sign on the diffusion term, which is why we use abs(Drr) here. 
+                // This should however probably be reworked in a better way...
+                if(!isNegativeTrappedRadial && ( abs(Drr(ir, i, j, drr)) || abs(Drr(ir+1, i, j, drr)) ) ){
```

Comment:
I don't quite follow the logic here... To me it seems that the ``abs(...)`` checks if the diffusion coefficient is non-zero, which could equivalently (and more clearly and safely) be done with ``Drr(...) != 0``. I also don't quite understand which situation the comment above refers to? Would it happen in a case where the coefficient provided by the user is still "correct"? Or is this a way of changing the sign of a diffusion coefficient which was given the "wrong" sign by the user?

Reply:
Well, the `Drr` has to be negative for the ions due to the IonTransientTerm having a different sign than the ordinary TransientTerm, while for other quantities `Drr` should be positive. (the user still gives positive coefficents in the python interface, and then `Drr` is set to minus the user-given value). But as we don't know here what quantity `Drr` applies to, we need to accept both positive and negative `Drr`. And yes, the logic would be the same if the `abs()` were replaced by `Drr(...) != 0`.

Reply:
Ok... But so it doesn't really have to with the sign of ``Drr``, but just whether it's non-zero or not?

When taking the absolute value in C++, you should make a habit of explicitly writing ``std::abs()`` since you might otherwise get the C version of ``abs()`` without the compiler telling you. The C version only operates on integers, so in this case any diffusion coefficient with ``|Drr| < 1`` would be treated as identically zero (assuming you are unlucky and get the C version of ``abs()``). I would still highly recommend checking for zero explicitly instead, since it's both clearer and faster.